Sources/TensorFlow/Core/LazyTensorOperation.swift

```diff
@@ -130,6 +130,57 @@ class LazyTensor: _AnyTensorHandle {
     static var _materializationCallback: (String) -> () = { _ in }
 }
 
+extension _AnyTensorHandle {
+    /// Returns a concrete `LazyTensor` with an additional constraint that the
+    /// underlying concrete `LazyTensor` should be marked to be promoted as an
+    /// input when used in an extracted trace.  This provides a **temporary**
+    /// mechanism to promote a concrete lazy tensor to an input in extracted
+    /// traces. (Note that this may trigger materialization.)
+    var _concreteInputLazyTensor: LazyTensor {
+        LazyTensor(_materialized: self._tfeTensorHandle)
+    }
+}
+
+extension TensorHandle {
+    /// Returns `Self` that wraps `_concreteInputLazyTensor` of the underlying
+    /// `_AnyTensorHandle`
+    public var _concreteInputLazyTensor: TensorHandle {
```

Comment:
Hmm is there a reason this needs to be public?

Reply:
I expect that it will only be used within the library. However, I wanted it to be available for experimentation in the user models if needed.

Reply:
How and why should a user be expected to use this?

Reply:
Just wanted to be careful exposing public APIs. I think we need to make sure public APIs have at least some justification of use cases before they are exposed.

Reply:
The use case is the one in description where we want to force a constant to be an input to an extracted trace. Ultimately, we may not need this API. However, in the meantime,  I want to have this knob while we develop and improve the lazy tensor implementation.

Reply:
Oh I see. Sorry for missing parts of the PR description! And thanks for explaining :)